packages/design-system/src/components/Box/Box.tsx

```diff
@@ -321,6 +330,38 @@ const StyledBox = styled.div<PropsToTransientProps<TransientBoxProps>>`
 
   // Cursor
   cursor: ${({ $cursor }) => $cursor};
+
+  ${({ theme, ...props }) => {
+    // Get only the responsive values from the props
+    const responsiveProps = ((props) => ({
+      padding: props.$padding,
+      paddingTop: props.$paddingTop,
+      paddingBottom: props.$paddingBottom,
+      paddingLeft: props.$paddingLeft,
+      paddingRight: props.$paddingRight,
+      margin: props.$margin,
+      marginTop: props.$marginTop,
+      marginBottom: props.$marginBottom,
+      marginLeft: props.$marginLeft,
+      marginRight: props.$marginRight,
+      gap: props.$gap,
+      color: generateResponsiveValues(theme.colors, props.$color),
+      background: generateResponsiveValues(theme.colors, props.$background),
+      'font-size': generateResponsiveValues(theme.fontSizes, props.$fontSize),
+      'border-radius': props.$hasRadius ? theme.borderRadius : props.$borderRadius,
+      'border-style': props.$borderStyle,
+      'border-width': props.$borderWidth,
+      'border-color': generateResponsiveValues(theme.colors, props.$borderColor),
+      border: getBorderProp({
+        theme,
+        $borderColor: props.$borderColor,
+        $borderStyle: props.$borderStyle,
+        $borderWidth: props.$borderWidth,
+      }),
```

Comment:
this is confusing, in `handleResponsiveValues` you extract values from the spaces object of the theme but here you handle other props like colors. I think you should be doing them in the same place everywhere either in this function of the other to be consistent. You could alternatively, create a pipe function where you:
1. get the value from the theme or use the value provided (to allow CSS values)
2. pass to the repsonsive function to extract and create the media queries
food for thought.

Reply:
Yes, I realised it yesterday and I will generate responsive values at single place. 
With pipe wont it be separate media query per prop which we dont want?

Reply:
yeah you're probably right.